### PR TITLE
Improved ValueError statements for verify_in_list and verify_same_elements

### LINKED ISSUE #494: Make error messages in verify_in_list and verify_same_elements more clear
**Is your feature request related to a problem? Please describe.**

The wording of the error messages for `misc_utils.verify_in_list` and `misc_utils.verify_same_elements` are not exactly clear. We should clarify the wording so the user understands clearly what is going on.

UPDATE: we also need to truncate the error messages of both functions if the lists contain a lot of mismatches.

**Describe the solution you'd like**

Update `err_msg` in both functions.

UPDATE: to handle error message truncation, limit to just printing the unique element failures for `misc_utils.verify_in_list`, and for both, limit to 10 incorrect values (or some reasonable value) displayed for one list.

### PR DESCRIPTION
**Pull request by srivarra**
**If you haven't already, please read through our [contributing guidelines](https://ark-analysis.readthedocs.io/en/latest/_rtd/contributing.html) before opening your PR**

**What is the purpose of this PR?**

Closes #494. The purpose of this PR is to improve the user readability of error messages in `misc_utils::verify_in_list` and `misc_utils::verify_same_elements`.

**How did you implement your changes**

Adjusted `misc_utils::verify_in_list` and `misc_utils::verify_same_elements` by allowing up to 10 invalid values to be displayed in the `ValueError` print statement. In addition created a new function `misc_utils::create_invalid_data_str` which returns a formatted string for printing these invalid values. There is also a test for this function.

**Remaining issues**

None.

### CODE REVIEW

File: ark/utils/misc_utils_test.py [hunk 1]
```diff
@@ -91,3 +91,29 @@ def test_verify_same_elements():
...
+
+def test_create_invalid_data_str():
+    invalid_data = ["data_" + str(i) for i in range(20)]
+
+    # Test to make sure the case of 10 invalid values creates a proper string.
+    invalid_data_str1 = misc_utils.create_invalid_data_str(invalid_data=invalid_data[:10])
+    result1 = []
+    for id in invalid_data[:10]:
+        result1.append(invalid_data_str1.find(id))
+    assert -1 not in result1
```
Comment (alex-l-kong) [inline ark/utils/misc_utils_test.py:104]: You should also assert that `result1 = list(range(10))`

File: ark/utils/misc_utils_test.py [hunk 2]
```diff
@@ -91,3 +91,29 @@ def test_verify_same_elements():
...
+
+    # Test to make sure cases of less than 10 invalid values creates a proper string.
+    invalid_data_str2 = misc_utils.create_invalid_data_str(invalid_data=invalid_data[:3])
+    result2 = []
+    for id in invalid_data[:3]:
+        result2.append(invalid_data_str2.find(id))
+    assert -1 not in result2
```
Comment (alex-l-kong) [inline ark/utils/misc_utils_test.py:111]: Similar comment as above except for `list(range(3))`

File: ark/utils/misc_utils_test.py [hunk 3]
```diff
@@ -91,3 +91,29 @@ def test_verify_same_elements():
...
+
+    # Test to make sure cases of more than 10 invalid values creates a proper string
+    # capping out at 10 values.
+    invalid_data_str3 = misc_utils.create_invalid_data_str(invalid_data=invalid_data)
+    result3 = []
+    for id in invalid_data[:10]:
+        result3.append(invalid_data_str3.find(id))
+    assert -1 not in result3
```
Comment (alex-l-kong) [inline ark/utils/misc_utils_test.py:119]: Test equality with `list(range(10))`

**Review by alex-l-kong — CHANGES_REQUESTED** [review]
The string formatting looks a lot better, mostly comments about testing.

File: ark/utils/misc_utils.py [hunk 4]
```diff
@@ -117,14 +117,42 @@ def save_figure(save_dir, save_file, dpi=None):
     plt.savefig(os.path.join(save_dir, save_file), dpi=dpi)
 
 
+def create_invalid_data_str(invalid_data):
+    """Creates a easy to read string for ValueError statements.
+
+    _extended_summary_
```
[the remark this answers is not in the document]
Comment (srivarra) [reply ark/utils/misc_utils.py:123]: Ah, it's an artifact of a vs code extension that creates a template for functions and classes. I'll remove it.

File: ark/utils/misc_utils_test.py (hunk 1 above)
Comment (srivarra) [reply ark/utils/misc_utils_test.py:104]: `.find()` returns the first index where the search string is found, and $-1$ if it cannot find it. So its testing to make sure that up to the first 10 invalid values are in the print statement by checking if there is not a $-1$ in the `result1` list.

I could also assert that the length of the `result1` list is the same as the lengths of the `invalid_data` used.

However I'm not too sure what the assert for `result1 == list(range(10))` would mean.

Comment (alex-l-kong) [reply ark/utils/misc_utils_test.py:104]: My bad, I completely missed your use of `.find`, somehow thought you were appending the raw values to `result1`. This works, you can ignore the rest of the comments I made.

Comment (alex-l-kong) [reply ark/utils/misc_utils_test.py:104]: Alternatively:

```
for id in invalid_data[:10]:
    assert invalid_data_str1.find(id) != -1
```

**Review by ngreenwald — APPROVED** [review]
Nothing to add on top of Alex, looks good

File: ark/utils/misc_utils.py [hunk 5]
```diff
@@ -137,15 +163,22 @@ def verify_in_list(**kwargs):
...
 
-        raise ValueError(err_str % (test_list_name, bad_vals, good_values_name))
+        # Only printing up to the first 10 invalid values.
+        err_str_1 = ("Displaying {0} of {1} invalid value(s) provided for list {2:^}.\n").format(
+            min(len(difference), 10), len(difference), test_list_name
+        )
+
+        err_str_3 = create_invalid_data_str(difference)
+
+        err_str = err_str_1 + err_str_3
```
Comment (alex-l-kong) [inline ark/utils/misc_utils.py:180]: The names `err_str_1` and `err_str_3` should be more descriptive. You could rename each one of these, but I personally prefer to just have one `err_str` and append each subsequent update:

```
err_str = ("Displaying {0} of {1} invalid value(s) provided for list {2:^}.\n").format(
    min(len(difference), 10), len(difference), test_list_name
)

err_str += create_invalid_data_str(difference)
```

File: ark/utils/misc_utils.py [hunk 6]
```diff
@@ -172,12 +205,39 @@ def verify_same_elements(**kwargs):
...
+        err_str_4 = ("{0:>13} \n").format(
+            "Displaying {0} of {1} missing value(s) for list {2}\n".format(
+                min(len(missing_vals_2), 10), len(missing_vals_2), list_two_name
+            )
+        )
+        err_str_5 = create_invalid_data_str(missing_vals_2) + "\n"
 
-        err_str = ("Invalid value(s) provided in both %s and %s variables: value(s)"
-                   " %s not found in both lists")
+        err_str = err_str_1 + err_str_2 + err_str_3 + err_str_4 + err_str_5
```
Comment (alex-l-kong) [inline ark/utils/misc_utils.py:241]: Similar comment regarding `err_str` naming.

**Review by alex-l-kong — CHANGES_REQUESTED** [review]
A couple comments related to variable naming and the creation of `err_str`.

**Review by alex-l-kong — APPROVED** [review]
Looks good!

**Pull request merged**